pkg/scale/encode.go

```diff
@@ -56,32 +57,68 @@ func (es *encodeState) marshal(in interface{}) (err error) {
 	switch in := in.(type) {
 	case int:
 		err = es.encodeUint(uint(in))
+		if err != nil {
+			err = fmt.Errorf("encoding Uint: %w", err)
+		}
 	case uint:
 		err = es.encodeUint(in)
+		if err != nil {
+			err = fmt.Errorf("encoding Uint: %w", err)
+		}
 	case int8, uint8, int16, uint16, int32, uint32, int64, uint64:
 		err = es.encodeFixedWidthInt(in)
+		if err != nil {
+			err = fmt.Errorf("encoding FixedWidthInt: %w", err)
```

Comment:
supernit no need to use the CamelCaseNameOfTheMethod 😉  But feel free to leave as is, it's fine as well!

```suggestion
			err = fmt.Errorf("encoding fixed width int: %w", err)
```